Storage/tests/System/ManageObjectsTest.php

```diff
@@ -414,4 +444,21 @@ public function testStringNormalization()
             $this->assertSame($expectedContent, $actualContent);
         }
     }
+
+    private function assertStorageObjectExists($object, $options = [], $isPresent = true)
+    {
+        $this->assertEquals($isPresent, $object->exists($options));
+        $object = self::$bucket->object($object->name(), $options);
+        $this->assertEquals($isPresent, $object->exists($options));
+        $objects = self::$bucket->objects($options);
+        $objects = array_map(function ($o) {
+            return $o->name();
+        }, iterator_to_array($objects));
+        $this->assertEquals($isPresent, in_array($object->name(), $objects));
```

Comment:
Why do we need these many tests? Just the `$this->assertEquals($isPresent, $object->exists($options));` is not sufficient enough?

Reply:
Yes these are three tests which asserts the existence on three types of objects:
1. provided `$object` instance
2. `$bucket::object(..)` instance
3. `$bucket::objects(..)` should contain object under test

Reply:
sounds good, thanks